### PR TITLE
Embed and animation visibility

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: extensions/amp-animation/0.1/amp-animation.js [hunk 1]
```diff
@@ -42,17 +60,125 @@ export class AmpAnimation extends AMP.BaseElement {
...
     // Parse config.
     const scriptElement = user().assert(
         childElementByTag(this.element, 'script'),
         '"<script type=application/json>" must be present');
     this.configJson_ = tryParseJson(scriptElement.textContent, error => {
       throw user().createError('failed to parse animation script', error);
     });
+
+    if (this.triggerOnVisibility_) {
+      setStyles(this.element, {
```
Comment (cramforce) [inline extensions/amp-animation/0.1/amp-animation.js]: Why? Please comment.

File: extensions/amp-animation/0.1/amp-animation.js [hunk 2]
```diff
@@ -42,17 +60,125 @@ export class AmpAnimation extends AMP.BaseElement {
     user().assert(isExperimentOn(this.win, TAG),
         `Experiment "${TAG}" is disabled.`);
 
+    // Trigger.
+    const trigger = this.element.getAttribute('trigger');
+    if (trigger) {
+      this.triggerOnVisibility_ = user().assert(
+          trigger == 'visibility',
```
Comment (cramforce) [inline extensions/amp-animation/0.1/amp-animation.js]: I'm a bit confused by what `visibility` means here and what the goal of the PR is. Is this `document-visibility`?

File: extensions/amp-animation/0.1/amp-animation.js [hunk 3]
```diff
@@ -42,17 +60,125 @@ export class AmpAnimation extends AMP.BaseElement {
...
+        height: '1px',
+        display: 'block',
+        position: 'fixed',
+      });
+    }
+
+    // Visibility.
+    const ampdoc = this.getAmpDoc();
+    const frameElement = getParentWindowFrameElement(this.element, ampdoc.win);
+    const embed =
```
Comment (cramforce) [inline extensions/amp-animation/0.1/amp-animation.js]: Can we at all avoid that `amp-animation` needs to be aware of being embedded?

File: extensions/amp-animation/0.1/amp-animation.js (hunk 2 above)
Comment (cramforce) [inline extensions/amp-animation/0.1/amp-animation.js]: So, it seems for embeds it is driven by the container `viewportCallback`, but more non-embedded it is just doc visibility?

Comment (dvoytenko) [inline extensions/amp-animation/0.1/amp-animation.js]: As discussed offline, we will extend the visibility to the direct parent, but for now we will keep it defined as `:root` == embed visibility. As such `amp-animation` is restricted to only be placed directly inside `<body>` element.

File: extensions/amp-animation/0.1/amp-animation.js (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-animation/0.1/amp-animation.js]: Done. Our normal hack to ensure that `layoutCallback` is called.

File: extensions/amp-animation/0.1/amp-animation.js (hunk 3 above)
Comment (dvoytenko) [inline extensions/amp-animation/0.1/amp-animation.js]: Working on this across FIE, in-a-box and main doc. WIP.

**Comment by dvoytenko** [comment]
@cramforce Comments addressed. PTAL.

File: extensions/amp-animation/amp-animation.md [hunk 4]
```diff
@@ -234,12 +239,25 @@ can be reduced to an array of components. For instance:
...
+The animation can be triggered via a `trigger` attribute an `on` action.
+
+### `trigger` attribute
 
+For instance:
 ```
-<amp-animation id="anim1" layout="nodisplay"></amp-animation>
-<button on="tap:anim1.activate">Animate</button>
+<amp-animation id="anim1" layout="nodisplay"
+    trigger="visibility">
```
Comment (cramforce) [inline extensions/amp-animation/amp-animation.md]: Do you want to document the actual value?

Comment (dvoytenko) [inline extensions/amp-animation/amp-animation.md]: Done

**Pull request merged**